lib/resources/csv.rb

```diff
@@ -45,8 +47,10 @@ def value(key)
       @params.map { |x| x[key.first.to_s] }.compact
     end
 
-    def to_s
-      "Csv #{@path}"
+    # used by JsonConfig to build up a full to_s method
+    # based on whether a file path, content, or command was supplied.
+    def resource_base_name
```

Comment:
should this be a private function then?

Reply:
I've historically had issues with `to_s` using private methods. Perhaps it's fixed in later Ruby versions. I'll change it and we can see what happens.

Reply:
to_s needs to be public, i am just taling about the `resource_base_name`